### ISSUE TITLE
Ability to not leave a comment on `terraform plan` if nothing changed

### ISSUE THREAD

**Issue by alex** [hashicorp/terraform-github-actions#29]
I've got my terraform resources in a monorepo. As a result of this, lots of commits do not touch terraform at all, and terraform plan leaving a comment on all of them doesn't add much value (and is confusing and distracting for devs working on other things).

A configuration option to only leave a comment if `terraform plan` detects work to do would be fantastic. Thanks!

**Comment by epechuzal** [comment]
While you wait for a reply on this, maybe this will work for you:

Filter your workflow yml to only trigger when the terraform paths are changed, e.g.
```
on:
  pull_request:
    paths:
    - "path/to/terraform/*"
    - "path/to/terraform/*/*"
    - "path/to/terraform/*/*/*"
```
The last two lines are to catch any changes to files in subdirs. Depending on how you set up your tf files, you might not need those.

**Comment by sudomateo** [comment]
@alex This might be possible by using the `-detailed-exitcode` option to `terraform plan`.

```
  -detailed-exitcode  Return detailed exit codes when the command exits. This
                      will change the meaning of exit codes to:
                      0 - Succeeded, diff is empty (no changes)
                      1 - Errored
                      2 - Succeeded, there is a diff
```

Given this and some other issues, I'm rethinking the way this delivers comments to the pull request, since a very large comment can either be rejected by the API, cause an error via `jq`, or just reduce readability in the pull request.

**Comment by sudomateo** [comment]
Should be fixed in #88.
